### PR TITLE
Prototypal inheritance

### PR DESCRIPTION
**Pull request by flaviohblima**
Prototypal inheritance

### CODE REVIEW

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 1]
```diff
@@ -306,35 +306,35 @@ for(let prop in rabbit) {
...
+...Mas por que `hasOwnProperty` não aparece no loop `for..in`, da mesma forma que `eats` e `jumps`, já que o `for..in` lista propriedades herdadas?
 
-The answer is simple: it's not enumerable. Just like all other properties of `Object.prototype`, it has `enumerable:false` flag. And `for..in` only lists enumerable properties. That's why it and the rest of the `Object.prototype` properties are not listed.
+A resposta é simples: ele não é enumerável. Assim como todas as outras propriedades de `Object.prototype`, ele tem a flag `enumerable:false`. E `for..in` só lista aquelas que são enumeráveis. Aí está o porquê do resto das propriedades de `Object.prototype` não estarem listadas.
 
-```smart header="Almost all other key/value-getting methods ignore inherited properties"
-Almost all other key/value-getting methods, such as `Object.keys`, `Object.values` and so on ignore inherited properties.
+```smart header="Quase todas as outras chaves/métodos *value-gettting* ignoram propriedades herdadas."
+Quase todas as outras chaves/métodos *value-gettting*, como `Object.keys`, `Object.values`, entre outros, ignoram propriedades herdadas.
```
Comment (odsantos) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:327]:
```suggestion
Quase todas as outras chaves/métodos *obtém-valor*, como `Object.keys`, `Object.values`, entre outros, ignoram propriedades herdadas.
```

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 2]
```diff
@@ -306,35 +306,35 @@ for(let prop in rabbit) {
...
 
-- In JavaScript, all objects have a hidden `[[Prototype]]` property that's either another object or `null`.
-- We can use `obj.__proto__` to access it (a historical getter/setter, there are other ways, to be covered soon).
-- The object referenced by `[[Prototype]]` is called a "prototype".
-- If we want to read a property of `obj` or call a method, and it doesn't exist, then JavaScript tries to find it in the prototype.
-- Write/delete operations act directly on the object, they don't use the prototype (assuming it's a data property, not a setter).
-- If we call `obj.method()`, and the `method` is taken from the prototype, `this` still references `obj`. So methods always work with the current object even if they are inherited.
-- The `for..in` loop iterates over both its own and its inherited properties. All other key/value-getting methods only operate on the object itself.
+- No Javascript, todos os objetos possuem uma propriedade `[[Prototype]]` escondida que ou é um objeto ou é `null`.
+- Nós podemos usar `obj.__proto__` para acessá-lo (um getter/setter que é histórico, para o qual já temos alternativas, que veremos em breve).
```
Comment (odsantos) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:335]:
```suggestion
- Nós podemos usar `obj.__proto__` para acessá-lo (um histórico getter/setter, mas já existem alternativas, que veremos em breve).
```

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 3]
```diff
@@ -306,35 +306,35 @@ for(let prop in rabbit) {
...
+- O objeto referenciado por `[[Prototype]]` é um chamado "protótipo" (*prototype*).
+- Se nós queremos ler uma propriedade de `obj` ou chamar um método, e eles não existem, o Javascript tenta encontrá-los no protótipo.
+- Operações de escrever/deletar agem diretamente no objeto, elas não usam o protótipo (assumindo que a propriedade seja um dado, não um setter).
```
Comment (odsantos) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:338]:
```suggestion
- Operações de escrever/deletar agem diretamente no objeto, elas não usam o protótipo (assumindo que seja uma propriedade para dados, não um setter).
```

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 4]
```diff
@@ -306,35 +306,35 @@ for(let prop in rabbit) {
...
+- Se chamarmos `obj.method()`, e o `method` é obtido do prototype, o `this` ainda referencia o `obj`. Portanto métodos sempre trabalham com o objeto atual, mesmo que eles sejam herdados.
+- O loop `for..in` itera tanto sobre as propriedades do objeto quanto sobre as propriedades herdadas. Todas as outras chaves/métodos *value-getting* operam apenas sobre o próprio objeto.
```
Comment (odsantos) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:340]:
```suggestion
- O loop `for..in` itera tanto sobre as propriedades do objeto quanto sobre as propriedades herdadas. Todas os outros métodos  para obter chaves/valores operam apenas sobre o próprio objeto.
```

**Review by odsantos — CHANGES_REQUESTED** [review]
Hello @flaviohblima ,

Here is a review.
Please, feel free to leave a note where you might disagree.

Thank you,
Osvaldo.

**Comment by odsantos** [comment]
Hi @flaviohblima ,

Still working in this one?
There is a conflict here.

Please, let me know if you haven't time and I might be able to resolve it.

Thanks,
Osvaldo.

**Comment by flaviohblima** [comment]
Hi @odsantos ,

Sorry I could not resolve this pull request.
I might work in some translations soon.

In any case, feel free to resolve the conflicts and close it.

Team work is always quickier!

Thank you,
Flávio.

**Comment by odsantos** [comment]
Hi @flaviohblima ,

I'll try to resolve the conflicts but keep it open.
Probably, it can be reviewed by others maintainers, as well, and get merged.

Thank you,
Osvaldo.

**Comment by nazarepiedady** [comment]
@Peruibeloko, @jonnathan-ls, do you have time to see what is possible to do in relation to this pull request?

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 5]
```diff
@@ -1,22 +1,22 @@
-# Prototypal inheritance
+# Herança Prototipada
 
-In programming, we often want to take something and extend it.
+Na programação, nós sempre queremos extender algum comportamento.
```
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:3]:
```suggestion
Na programação, é comum querermos extender algum comportamento.
```

File: 1-js/08-prototypes/01-prototype-inheritance/1-property-after-delete/solution.md [hunk 6]
```diff
@@ -1,4 +1,4 @@
 
-1. `true`, taken from `rabbit`.
-2. `null`, taken from `animal`.
-3. `undefined`, there's no such property any more.
+1. `true`, obtido de `rabbit`.
+2. `null`, obtido de `animal`.
+3. `undefined`, essa propriedade não existe mais.
```
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/1-property-after-delete/solution.md:4]:
```suggestion
1. `true`, obtido de `coelho`.
2. `null`, obtido de `animal`.
3. `undefined`, essa propriedade não existe mais.
```

File: 1-js/08-prototypes/01-prototype-inheritance/1-property-after-delete/task.md
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/1-property-after-delete/task.md:1]: Traduzir as variáveis

File: 1-js/08-prototypes/01-prototype-inheritance/2-search-algorithm/solution.md
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/2-search-algorithm/solution.md:1]: Traduzir o código

File: 1-js/08-prototypes/01-prototype-inheritance/2-search-algorithm/task.md
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/2-search-algorithm/task.md:1]: Traduzir o código

File: 1-js/08-prototypes/01-prototype-inheritance/3-proto-and-this/solution.md
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/3-proto-and-this/solution.md:1]: Traduzir as variáveis

File: 1-js/08-prototypes/01-prototype-inheritance/3-proto-and-this/task.md
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/3-proto-and-this/task.md:1]: Traduzir o código

File: 1-js/08-prototypes/01-prototype-inheritance/4-hamster-proto/solution.md
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/4-hamster-proto/solution.md:1]: Traduzir o código

File: 1-js/08-prototypes/01-prototype-inheritance/4-hamster-proto/task.md
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/4-hamster-proto/task.md:1]: Traduzir o código

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 7]
```diff
@@ -1,22 +1,22 @@
...
 
-For instance, we have a `user` object with its properties and methods, and want to make `admin` and `guest` as slightly modified variants of it. We'd like to reuse what we have in `user`, not copy/reimplement its methods, just build a new object on top of it.
+Por exemplo, nós temos um objeto `user` com suas propriedades e métodos, e queremos criar duas variantes `admin` e `guest` ligeiramente diferentes. Nós gostaríamos de reutilizar o que já temos no `user`, sem copiar/reimplementar seus métodos, apenas construir um novo objeto baseado nele.
```
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:5]:
```suggestion
Por exemplo, nós temos um objeto `usuario` com suas propriedades e métodos, e queremos criar duas variantes `admin` e `convidado` ligeiramente diferentes. Nós gostaríamos de reutilizar o que já temos no `usuario`, sem copiar/reimplementar seus métodos, apenas construir um novo objeto baseado nele.
```

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 8]
```diff
@@ -1,22 +1,22 @@
...
-When we read a property from `object`, and it's missing, JavaScript automatically takes it from the prototype. In programming, this is called "prototypal inheritance". And soon we'll study many examples of such inheritance, as well as cooler language features built upon it.
+Quando lemos uma propriedade de um `object`, e ela não está presente, o Javascript automaticamente obtém seu valor do protótipo. Na programação, isso é chamado de "herança prototipada". Em breve nós vamos ver vários exemplos dessa herança, bem como funcionalidades legais construidas em cima disso.
 
-The property `[[Prototype]]` is internal and hidden, but there are many ways to set it.
+A propriedade `[[Prototype]]` é interna e escondida, mas existem várias formas de configurar um valor para ela. 
 
-One of them is to use the special name `__proto__`, like this:
+Uma delas é usar o nome especial `__proto__`, dessa forma:
 
 ```js run
```
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:21]: Traduzir o código

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 9]
```diff
@@ -115,47 +115,47 @@ let longEar = {
...
 
-Now if we read something from `longEar`, and it's missing, JavaScript will look for it in `rabbit`, and then in `animal`.
+Agora, se nós quisermos ler algo de `longEar` e ele não possuir, o Javascript vai procurar no `rabbit`, e depois no `animal`.
 
-There are only two limitations:
+Só existem duas limitações:
 
-1. The references can't go in circles. JavaScript will throw an error if we try to assign `__proto__` in a circle.
-2. The value of `__proto__` can be either an object or `null`. Other types are ignored.
+1. As referências não podem ser circulares. O Javascript vai disparar um erro caso a gente tente atribuir `__proto__` em um círculo de objetos.
```
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:129]:
```suggestion
1. As referências não podem ser circulares. O Javascript vai disparar um erro caso a gente tente atribuir `__proto__` ao seu próprio objeto, ou ao objeto que o utiliza, de maneira recursiva.
```

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 10]
```diff
@@ -115,47 +115,47 @@ let longEar = {
...
-The prototype is only used for reading properties.
+O protótipo é usado apenas para leitura de propriedades.
 
-Write/delete operations work directly with the object.
+As operações de escrita/deleção trabalham diretamente com o objeto.
 
-In the example below, we assign its own `walk` method to `rabbit`:
+No exemplo abaixo, nós criamos um método `walk` próprio para o `rabbit`:
 
 ```js run
```
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:154]: Traduzir o restante do código

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 11]
```diff
@@ -165,20 +165,20 @@ let rabbit = {
...
 
 ![](proto-animal-rabbit-walk-2.svg)
 
-Accessor properties are an exception, as assignment is handled by a setter function. So writing to such a property is actually the same as calling a function.
+Métodos de acesso são uma exceção, porque a atribuição é feita por uma função setter. Assim, guardar um valor numa dessas propriedades é o mesmo que chamar uma função.
 
-For that reason `admin.fullName` works correctly in the code below:
+Por essa razão, `admin.fullName` funciona corretamente no código abaixo:
 
 ```js run
```
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:183]: Traduzir o código

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 12]
```diff
@@ -201,37 +201,37 @@ let admin = {
...
-That is actually a super-important thing, because we may have a big object with many methods, and have objects that inherit from it. And when the inheriting objects run the inherited methods, they will modify only their own states, not the state of the big object.
+Na verdade isso é super importante, porque pode ser que a gente tenha um objeto grande com vários métodos, sendo que outros objetos herdam ele. Quando os objetos herdeiros rodam os métodos herdados, eles vão modificar apenas seus próprios estados, não o estado do objeto grande.
 
-For instance, here `animal` represents a "method storage", and `rabbit` makes use of it.
+Por exemplo, aqui `animal` representa um "armazenador de métodos", e `rabbit` faz uso deles.
 
-The call `rabbit.sleep()` sets `this.isSleeping` on the `rabbit` object:
+A chamada de `rabbit.sleep()` configura `this.isSleeping` no objeto `rabbit`:
 
 ```js run
```
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:229]: Traduzir o resto do código

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 13]
```diff
@@ -240,30 +240,30 @@ let animal = {
...
-## for..in loop
+## Loop for..in
 
-The `for..in` loop iterates over inherited properties too.
+O loop `for..in` também itera sobre as propriedades herdadas.
 
-For instance:
+Por exemplo:
 
 ```js run
```
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:268]: Traduzir o código

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 14]
```diff
@@ -304,35 +304,35 @@ for(let prop in rabbit) {
...
 
-...But why does `hasOwnProperty` not appear in the `for..in` loop like `eats` and `jumps` do, if `for..in` lists inherited properties?
+...Mas por que `hasOwnProperty` não aparece no loop `for..in`, da mesma forma que `eats` e `jumps`, já que o `for..in` lista propriedades herdadas?
 
-The answer is simple: it's not enumerable. Just like all other properties of `Object.prototype`, it has `enumerable:false` flag. And `for..in` only lists enumerable properties. That's why it and the rest of the `Object.prototype` properties are not listed.
+A resposta é simples: ele não é enumerável. Assim como todas as outras propriedades de `Object.prototype`, ele tem a flag `enumerable:false`. E `for..in` só lista aquelas que são enumeráveis. Aí está o porquê do resto das propriedades de `Object.prototype` não estarem listadas.
 
-```smart header="Almost all other key/value-getting methods ignore inherited properties"
-Almost all other key/value-getting methods, such as `Object.keys`, `Object.values` and so on ignore inherited properties.
+```smart header="Quase todas as outras chaves/métodos *obtém-valor* ignoram propriedades herdadas."
```
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:324]:
```suggestion
```smart header="Quase todos os métodos que obtém valores a partir de chaves ignoram propriedades herdadas."
```

File: 1-js/08-prototypes/01-prototype-inheritance/article.md [hunk 15]
```diff
@@ -304,35 +304,35 @@ for(let prop in rabbit) {
...
 
-- In JavaScript, all objects have a hidden `[[Prototype]]` property that's either another object or `null`.
-- We can use `obj.__proto__` to access it (a historical getter/setter, there are other ways, to be covered soon).
-- The object referenced by `[[Prototype]]` is called a "prototype".
-- If we want to read a property of `obj` or call a method, and it doesn't exist, then JavaScript tries to find it in the prototype.
-- Write/delete operations act directly on the object, they don't use the prototype (assuming it's a data property, not a setter).
-- If we call `obj.method()`, and the `method` is taken from the prototype, `this` still references `obj`. So methods always work with the current object even if they are inherited.
-- The `for..in` loop iterates over both its own and its inherited properties. All other key/value-getting methods only operate on the object itself.
+- No Javascript, todos os objetos possuem uma propriedade `[[Prototype]]` escondida que ou é um objeto ou é `null`.
+- Nós podemos usar `obj.__proto__` para acessá-lo (um histórico getter/setter, mas já existem alternativas, que veremos em breve).
```
Comment (Peruibeloko) [inline 1-js/08-prototypes/01-prototype-inheritance/article.md:333]:
```suggestion
- Nós podemos usar `obj.__proto__` para acessá-lo (um getter/setter mantido historicamente, mas já existem alternativas, que veremos em breve).
```

**Review by Peruibeloko — CHANGES_REQUESTED** [review]
Parece bastante coisa, mas na verdade são só alterações pequenas e pedidos de tradução do código.

Acredito que seja importante traduzir os nomes usados nos exemplos do artigo, já que o tópico pode ser de difícil compreensão para iniciantes

**Comment by nazarepiedady** [comment]
@Peruibeloko, I was talking about to edit this pull request and approved if possible.

**Comment by Peruibeloko** [comment]
@nazarepiedady I believe the repo is write protected, but it I'll see what I can do

**Review by Peruibeloko — APPROVED** [review]

**Review by nazarepiedady — APPROVED** [review]

**Pull request merged**